app/src/lib/stores/app-store.ts

```diff
@@ -1466,6 +1468,71 @@ export class AppStore extends TypedBaseStore<IAppState> {
     }
   }
 
+  private detectMergeResolution(status: IStatusResult) {
+    const selection = this.getSelectedState()
+
+    if (selection === null || selection.type !== SelectionType.Repository) {
+      return
+    }
+
+    const { tip } = selection.state.branchesState
+
+    if (tip.kind !== TipState.Valid) {
+      return
+    }
+
+    const repository = selection.repository
+    const repoState = this.repositoryStateCache.get(repository)
+
+    // check if we're in a conflicted state
+    if (repoState.conflictState === null) {
+      return
+    }
+
+    const previousBranch = repoState.conflictState.branch
+    const currentBranchName = status.currentBranch
+
+    // no current branch means something happened so bail
```

Comment:
Many things that could happen to cause this, none of which we actually care about here. Distinguishing between them makes no difference at the moment because having a branch is an invariant in this function.

Reply:
I probably should add that _something_ could be good or bad, in either case, we don't care since having a branch selected is the thing we care about.